### DISCUSSION TITLE
app1#build dependsOn app2#dev

### DISCUSSION THREAD

**Discussion by ricardo-fnd** [vercel/turborepo#6806] [Help]
Hi. I want to build `router` app that needs `etl` app to be running already. So this is what I have:

![Screenshot 2023-12-15 at 14 44 51](https://github.com/vercel/turbo/assets/52207061/aa42e742-4a9e-4789-b707-b043c14b7537)

The problem with this is that `etl#dev` is a task that never finishes, so `router#build` never starts. I searched the docs, but no luck. Any suggestions? Thanks

**Comment by anthonyshew** [comment]
Hi, Ricardo. You're correct in that Turborepo isn't able to do this and why.

I'd be interested to learn more about your use case, though.
- What requirements do you have that make this pipeline necessary?
- Are you trying to trigger `router#build` while `etl#dev` is running? Is there meant to be a specific trigger by `etl#dev` that kicks off `router#build`?
- Whatever other details you'd like to share.

**Reply by ricardo-fnd** [reply]
HI @anthonyshew thanks for the reply.
[etl](https://en.wikipedia.org/wiki/Extract,_transform,_load) is an app named after a computing process that manipulates data.

For `router app` to be build, it needs some data. That's where `etl app` enters.
Yes. I'm trying to `build app router` while `elt` is in dev, so it can fetch the data.

What I do now is do this manually. `yarn dev` in one terminal and `yarn build` in other. And then to do `yarn start` I'll need to close the terminal with `yarn dev` so I can use the same ports.

**Reply by sohamnandi77** [reply]
Hi @ricardo-fnd  @anthonyshew, I also have similar use case, facing same problem. Please help me out.

In the below dev case, only @ace/helpers and @ace/api runs after that, it's stuck and @ace/ui or @ace/modules never start
![sample](https://github.com/vercel/turbo/assets/56152437/26885a98-dee5-419c-a023-ea30dee428b3)

![sample_build](https://github.com/vercel/turbo/assets/56152437/8dca3f67-351b-45ad-8b2b-a512fcb80d0d)
